### PR TITLE
Handle creds fetching for zonal vs regional gke clusters

### PR DESCRIPTION
**Pull request by michaeljguarino**
## Summary
You need to specify the zone to the cli for this, so use a dedicated context var to grab it

## Test Plan
regression tested a standard gcp workspace

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/provider/gcp.go [hunk 1]
```diff
@@ -270,6 +271,14 @@ func (gcp *GCPProvider) mkBucket(name string) error {
 	return nil
 }
 
+func (gcp *GCPProvider) clusterLocation() (string, string) {
+	if z, ok := gcp.ctx["clusterZone"]; ok {
```
Comment (DavidSpek) [inline pkg/provider/gcp.go:275]: Shouldn't we be setting this in the context somewhere?

Comment (michaeljguarino) [reply pkg/provider/gcp.go:275]: it has to be done manually, and we'll document it as part of the process

Comment (DavidSpek) [reply pkg/provider/gcp.go:277]: We should be able to handle that more gracefully but it’s fine for now.

**Review by DavidSpek — APPROVED** [review]
LGTM

**Pull request merged**